pilot/pkg/networking/core/v1alpha3/envoyfilter/cluster_patch.go

```diff
@@ -39,12 +40,66 @@ func ApplyClusterMerge(pctx networking.EnvoyFilter_PatchContext, efw *model.Envo
 			continue
 		}
 		if commonConditionMatch(pctx, cp) && clusterMatch(c, cp) {
-			proto.Merge(c, cp.Value)
+
+			ret, err := mergeTransportSocketCluster(c, cp)
+			if err != nil {
+				continue
+			}
+			if !ret {
+				proto.Merge(c, cp.Value)
+			}
 		}
 	}
 	return c
 }
 
+// Test if the patch contains a config for TransportSocket
+func mergeTransportSocketCluster(c *cluster.Cluster, cp *model.EnvoyFilterConfigPatchWrapper) (bool, error) {
+
+	err := fmt.Errorf("ERROR function mergeTransportSocketCluster failed.")
+
+	cpValueCast, okCpCast := (cp.Value).(*cluster.Cluster)
+	if !okCpCast {
+		log.Errorf("ERROR Cast of cp.Value: %v", okCpCast)
+		return false, err
+	}
+
+	var tsmPatch *cluster.Cluster_TransportSocketMatch
```

Comment:
We will generate config that uses TransportSocketMatch normally, but technically it can still also be in TransportSocket (if auto mtls is enabled). So we may need to handle that case as well